src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,52 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> previousNode;
+
+        public Node(Node previousNode, T value) {
+            this.previousNode = previousNode;
+            this.value = value;
+        }
+    }
 
+    public void push(T value) {
+        Node newNode = new Node(null, value);
+        if (tail == null) {
+            tail = newNode;
+        } else {
+            newNode.previousNode = tail;
+            tail = newNode;
+        }
```

Comment:
Do you really need this 'if' statement? And also, you can declare 'newNode.previousNode' in the constructor.

Reply:
You are rigth,anyway i make my Node tail and it not depends on exist tail or not. done